semanticdb/integration/src/main/java/com/javacp/Outer.java

```diff
@@ -0,0 +1,48 @@
+package com.javacp;
+
+class Outer {
+
+    class Inner {
+        // the compiler injects a first implicit param `this$` which is flagged mandated
+        // - only when using `-parameters` before JDK21
+        // - always as of JDK21
+        Inner(Integer... values) {}
+
+        public void printOuter() {
+            // For all JDKs, the generated code relies on an implicit class field `this$`
+            // holding a reference to the outer class.
+            System.out.println(Outer.this);
+        }
+
+        class InnerInner {
+            InnerInner(Integer... values) {}
+
+            public void printOuters() {
+                System.out.println(Outer.this);
```

Comment:
TIL: this reference is not captured directly, but indirectly through`Outer.Inner`, so there is only one implicit field & constructor param